pkg/leaderElection/leaderElection.go

```diff
@@ -0,0 +1,104 @@
+package leaderElection
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/datreeio/admission-webhook-datree/pkg/enums"
+	"github.com/datreeio/admission-webhook-datree/pkg/logger"
+	v1 "k8s.io/client-go/kubernetes/typed/coordination/v1"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+// official k8s example: https://github.com/kubernetes/client-go/blob/master/examples/leader-election
+
+type LeaderElection struct {
+	k8sClientLeaseGetter *v1.LeasesGetter
+	internalLogger       logger.Logger
+	isLeader             bool
+}
+
+func New(k8sClientLeaseGetter *v1.LeasesGetter, internalLogger logger.Logger) *LeaderElection {
+	if k8sClientLeaseGetter == nil {
+		internalLogger.LogAndReportUnexpectedError("leaderElection: k8s client is nil")
+		return &LeaderElection{
+			k8sClientLeaseGetter: nil,
+			internalLogger:       internalLogger,
+			isLeader:             true,
+		}
+	} else {
+		le := &LeaderElection{
+			k8sClientLeaseGetter: k8sClientLeaseGetter,
+			internalLogger:       internalLogger,
+			isLeader:             false,
+		}
+		// this function call is blocking, therefore we run it in a goroutine
+		go le.listenForChangesInLeader()
+		// block for 500ms to allow the first leader election to take place, in minikube it only takes 70ms
+		time.Sleep(500 * time.Millisecond)
+		return le
+	}
+}
+
+func (le *LeaderElection) IsLeader() bool {
+	return le.isLeader
+}
+
+func (le *LeaderElection) listenForChangesInLeader() {
+	uniquePodName := os.Getenv(enums.PodName)
+	namespace := os.Getenv(enums.Namespace)
+	if uniquePodName == "" {
+		panic(errors.New(fmt.Sprintf("env variable %s is not set", enums.PodName)))
+	}
+	if namespace == "" {
+		panic(errors.New(fmt.Sprintf("env variable %s is not set", enums.Namespace)))
+	}
```

Comment:
I think that in cases that we can't choose leader we need to use the default approach which means every pod is leader